### PR TITLE
fix: indicator does not appear while attaching onClick handler during guided tour

### PR DESCRIPTION
**Pull request by akash-codemonk**
## Description
This a scenario where the user might click on the explorer as it gets highlighted/takes focus from the user. After which when the user comes back the indicator isn't shown.
Pre fix

https://user-images.githubusercontent.com/67054171/151758778-835eccc3-5835-4099-8bdc-55ad5a9d14aa.mov

Post fix

https://user-images.githubusercontent.com/67054171/151758867-04f7bbdc-75af-4a42-abe1-77cd24c462f5.mov

To fix this we navigate the user to the canvas with the button widget selected when the user clicks on the cta.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Manually

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: fix/indicator-does-not-appear 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 55.74 **(0)** | 37.07 **(0.01)** | 35.8 **(0)** | 56.1 **(0)**
 :red_circle: | app/client/src/pages/Editor/GuidedTour/constants.tsx | 42.62 **(-0.48)** | 100 **(0)** | 4.55 **(-0.21)** | 42.62 **(-0.48)**
 :green_circle: | app/client/src/sagas/OnboardingSagas.ts | 22.22 **(0.32)** | 1.23 **(0)** | 4.35 **(0)** | 25 **(0.61)**
 :green_circle: | app/client/src/utils/WorkerUtil.ts | 89.76 **(0)** | 72.55 **(1.96)** | 100 **(0)** | 93.33 **(0)**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.94 **(0.23)** | 41.67 **(0.84)** | 36.21 **(0)** | 56.99 **(0.25)**</details>

### CODE REVIEW

File: app/client/src/pages/Editor/GuidedTour/constants.tsx [hunk 1]
```diff
@@ -497,11 +501,15 @@ export const Steps: StepsType = {
...
-          top: 25,
-          left: 0,
-        });
+        // Timeout is for a to wait for slight delay for the UI elements to appear
+        setTimeout(() => {
+          showIndicator(`[data-guided-tour-iid='onClick']`, "top", {
+            top: 25,
+            left: 0,
+          });
+        }, 1000);
```
Comment (arunvjn) [inline app/client/src/pages/Editor/GuidedTour/constants.tsx:512]: Can we do this without the timeout?

Comment (akash-codemonk) [reply app/client/src/pages/Editor/GuidedTour/constants.tsx]: using requestAnimationFrame instead

**Pull request merged**